### PR TITLE
Use `Message#mentions` instead of regex for leaderboard

### PR DESCRIPTION
**Pull request by suneettipirneni**
Discord.js has built-in support for mentions so we can use that instead, also in Discord.js v13 using a regex for mentions will make it harder to conform to discord.js types.

### CODE REVIEW

File: src/app/plugins/leaderboard.plugin.ts [hunk 1]
```diff
@@ -12,9 +12,7 @@ export class LeaderboardPlugin extends Plugin {
   public permission: ChannelType = ChannelType.Public;
   public pluginChannelName = Constants.Channels.Public.Games;
 
-  private readonly _MENTION_REGEX = /<@!?&?(\d+)>/;
-
-  public validate(message: IMessage, args: string[]) {
+  public validate(_message: IMessage, args: string[]) {
```
Comment (tanndlin) [inline src/app/plugins/leaderboard.plugin.ts:15]: why change `message` to `_message`

Comment (suneettipirneni) [reply src/app/plugins/leaderboard.plugin.ts:15]: In typescript you mark parameters you don't use by prefixing them with an underscore, this tells them to be skipped by the compiler.

**Review by tanndlin — APPROVED** [review]
I like red diffs

**Comment by suneettipirneni** [comment]
> I like red diffs

Ayyyyyy can I get one more approval on my PR: 🥺👉👈 😂😂😂 🥺👉👈

**Comment by TylerMathis** [comment]
I can't figure out how to approve this on mobile lol.

- [ ] Approved!

**Review by TylerMathis — APPROVED** [review]
KACHOW

**Pull request merged**